device-connectors/src/testflinger_device_connectors/devices/muxpi/muxpi.py

```diff
@@ -113,7 +113,53 @@ def _copy_to_control(self, local_file, remote_file):
             raise ProvisioningError(e.output)
         return output
 
+    def check_control_alive(self):
+        """Check if the control host is alive"""
+        try:
+            self._run_control("true")
+        except (ProvisioningError, subprocess.SubprocessError):
+            raise ProvisioningError(
+                "Control host is not responding, provisioning can't proceed!"
+            )
+
+    def reboot_control_host(self):
+        """
+        Reboot the control host
+        Sometimes the control host can end up in an unstable condition which
+        isn't easy to detect until it's too late. Since nothing else should
+        be using the control host, check and power cycle it if needed before
+        provisioning to ensure it's in a known good state.
+        """
+        if not self.config.get("control_host_reboot_script"):
+            logger.warning(
+                "control_host_reboot_script not defined, "
+                "skip rebooting control host"
+            )
+            return
+        logger.info("Rebooting control host")
+        for cmd in self.config["control_host_reboot_script"]:
+            logger.info("Running %s", cmd)
+            try:
+                subprocess.check_call(cmd.split(), timeout=60)
+            except Exception:
+                raise ProvisioningError("fail to reboot control host")
+
+        time.sleep(120)
+        # It should be up after 120s, but wait up to 5min if necessary
+        for _ in range(24):
```

Comment:
I also would like to suggest moving these numbers to a configuration file or at list file with constants, so we can update them easily in one place

Reply:
added `control_host_reboot_timeout` option in config file with default value set